### ISSUE TITLE
tsh ssh --identity does not use `ping` information

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#3513]
### Description

**What happened**:

`example.dev` and `1.2.3.4` used as an example here

```
➜  .tsh tsh ls --proxy ssh.example.dev:443 --login svc  --identity "../.ssh/svc.pem" -d
INFO             Reading identity file: ../.ssh/svc.pem common/identity.go:44
DEBU [TSH]       Extracted username "svc" from the identity file ../.ssh/svc.pem. tsh/tsh.go:945
INFO [CLIENT]    Connecting proxy=ssh.example.dev:3023 login='svc' method=0 client/api.go:1588
ERROR REPORT:
Original Error: *net.OpError dial tcp 1.2.3.4:3023: connect: operation timed out
Stack Trace:
Collapse
```

The proxy is correctly configured to provide the following ping information:

```
curl -k https://ssh.example.dev:443/webapi/ping
{"auth":{"type":"saml","second_factor":"off","saml":{"name":"OktaSAML","display":"SSO"}},"proxy":{"kube":{},"ssh":{"listen_addr":"0.0.0.0:3023","tunnel_listen_addr":"0.0.0.0:3024","public_addr":"ssh.example.dev:443","ssh_public_addr":"lb.example.dev:3023","ssh_tunnel_public_addr":"lb.example.dev:3024"}},"server_version":"4.2.6","min_client_version":"3.0.0"}
```

It seems that `tsh -i` does not use the `ssh_public_addr` information provided by ping, only `tsh ssh` uses it correctly.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`): 4.2.6
- Tsh version (use `tsh version`): 4.2.6
- OS (e.g. from `/etc/os-release`):

**Comment by webvictim** [comment]
I can confirm that this is a bug, only seen in cases where the `ssh_public_addr` is different to the `public_addr`.

You can work around it by providing the `ssh_public_addr` to `tsh` via the `-J` (jumphost) flag, like this:

```
$ tsh -J <ssh_public_addr>:3023 --proxy=ssh.example.dev:443 --identity=/.ssh/svc.pem --login=svc ls
```

You can get the `ssh_public_addr` with a command like this:

```
$ curl -s https://ssh.example.dev/webapi/ping | jq -r .proxy.ssh.ssh_public_addr
lb.example.dev:3023
```

You can combine this into one command like so:
```
$ tsh -J $(curl -s https://ssh.example.dev/webapi/ping | jq -r .proxy.ssh.ssh_public_addr) --proxy=ssh.example.dev:443 --identity=/.ssh/svc.pem --login=svc ls
```

**Comment by webvictim** [comment]
We should also fix #3032 at the same time as we fix this - it makes debugging potential issues much harder.

**Comment by awly** [comment]
A small clarification after reading the code here: commands that don't use the identity file don't use `Ping` RPC either.
They just have the info cached in user profile from the last `Login`.

I'm leaning towards only adding the `Ping` for the identity file code path and omit it for others. This way we don't add the RPC overhead to every command.
